src/rbs_string.c

```diff
@@ -5,61 +5,37 @@
 #include <stdio.h>
 #include <ctype.h>
 
-rbs_string_t rbs_string_shared_new(const char *start, const char *end) {
+rbs_string_t rbs_string_new(const char *start, const char *end) {
     return (rbs_string_t) {
         .start = start,
         .end = end,
-        .type = RBS_STRING_SHARED,
     };
 }
 
-rbs_string_t rbs_string_owned_new(const char *start, const char *end) {
-    return (rbs_string_t) {
-        .start = start,
-        .end = end,
-        .type = RBS_STRING_OWNED,
-    };
-}
-
-rbs_string_t rbs_string_copy_slice(rbs_string_t *self, size_t start_inset, size_t length) {
-    char *buffer = (char *) malloc(length + 1);
+rbs_string_t rbs_string_copy_slice(rbs_allocator_t *allocator, rbs_string_t *self, size_t start_inset, size_t length) {
+    char *buffer = rbs_allocator_calloc(allocator, length + 1, char);
```

Comment:
Follow-up: we're seeing a few of their cases where `calloc` is convenient for its API, but we don't need its behaviour `memset(0, ...)` (since we immediately overwrite it, here).

Perhaps we can add something like `rbs_allocator_alloc_many` which takes a count like `calloc` doesn, but _doesn't_ 0-out all the memory.